src/client.rs

```diff
@@ -113,20 +135,26 @@ impl Client<ElectrumPlaintextStream> {
 pub type ElectrumSslStream = SslStream<TcpStream>;
 #[cfg(feature = "use-openssl")]
 impl Client<ElectrumSslStream> {
-    /// Creates a new SSL client and tries to connect to `socket_addr`. Optionally, if `domain` is not
-    /// None, validates the server certificate.
-    pub fn new_ssl<A: ToSocketAddrs>(socket_addr: A, domain: Option<&str>) -> Result<Self, Error> {
+    /// Creates a new SSL client and tries to connect to `socket_addr`. Optionally, if
+    /// `validate_domain` is `true`, validate the server's certificate.
+    pub fn new_ssl<A: ToSocketAddrsDomain>(
+        socket_addr: A,
+        validate_domain: bool,
+    ) -> Result<Self, Error> {
         let mut builder =
             SslConnector::builder(SslMethod::tls()).map_err(Error::InvalidSslMethod)?;
         // TODO: support for certificate pinning
-        if domain.is_none() {
+        if !validate_domain {
```

Comment:
nit: inverting if branch to avoid the not

Reply:
fixed in 78a8a7c